### ISSUE TITLE
Device as argument only passes icon and name of device as parameters

### ISSUE THREAD

**Issue by StefanIndustries** [athombv/apps-sdk-issues#64]
In Homey 1.5 the complete device was passed on which made it possible to call the methods of the device. This is no longer possible because only the icon of the device and the name of it are passed on.

Is this intentional?

**Comment by WeeJeWel** [comment]
Please specify where this is happening. In a Flow, etc.

**Comment by StefanIndustries** [comment]
Issue is related to https://github.com/athombv/homey-apps-sdk-issues/issues/44

Basically, flows with action cards have the option to pass a device as argument.

On homey 1.5 this passed in the complete device class. Now this only passes on the icon of the device and the name. This makes it impossible to do something like this on Homey v2:
`
        {
          "id": "create_zone_with",
          "title": {
            "en": "Create zone with",
            "nl": "Maak zone met"
          },
          "fluidArgs": true,
          "args":[
            {
              "name": "device",
              "type": "device",
              "filter": "driver_id=soundtouch"
            },
            {
              "name": "slave",
              "type": "device",
              "placeholder": {
                "en": "Soundtouch device",
                "nl": "Soundtouch apparaat"
              },
              "filter": "driver_id=soundtouch"
            }
          ]
        },
`


```
        const _createZoneWithAction = new Homey.FlowCardAction('create_zone_with')
            .register()
            .registerRunListener(async (args, state) => {
                this.log('create zone', args);
                try {
                    await args.device._api.createZone(args.slave.getSettings().ip, args.slave.getSettings().mac);
                    return Promise.resolve(true);
                } catch (e) {
                    return promise.reject(e);
                }
            });
```

My devices have _api as one of their properties, but this is now no longer accessable.

Using these action cards now result in an error with message: Promise not resolved

**Comment by WeeJeWel** [comment]
That definitely doesn't sound right. Could you try to save the Flow using https://flow.homey.app and see if it also occurs there? This might be a mobile app related issue.

**Comment by StefanIndustries** [comment]
Same result. "Promise is not defined". Off-topic, that flow editor looks neat 👍

**Comment by WeeJeWel** [comment]
This has been fixed in Homey v2.0.5. Thank you for reporting!
